pkg/api/v1/types.go

```diff
@@ -0,0 +1,79 @@
+// +k8s:deepcopy-gen=package,register
+// +k8s:defaulter-gen=TypeMeta
+// +k8s:openapi-gen=true
+// +groupName=ingress.operator.openshift.io
+package v1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// DNSRecord represents a DNS record.
```

Comment:
How about a brief statement such as the following? "This resource kind is intended for internal use by the operator only."

Reply:
Would like to run this entire thing by API folks before code freeze. I'm not even sure if a v1 version is appropriate. We can tweak it.

Reply:
We could add something about how it is unstable too.  The idea is that anyone looking at the resource using `oc explain` or OpenShift Console would at least have some idea of what the thing is and know not to mess with it.